compiler/src/dotty/tools/dotc/core/Names.scala

```diff
@@ -546,13 +546,15 @@ object Names {
   private[this] var size = 1
 
   /** The hash of a name made of from characters cs[offset..offset+len-1].  */
-  private def hashValue(cs: Array[Char], offset: Int, len: Int): Int =
-    if (len > 0)
-      (len * (41 * 41 * 41) +
-        cs(offset) * (41 * 41) +
-        cs(offset + len - 1) * 41 +
-        cs(offset + (len >> 1)))
-    else 0
+  private def hashValue(cs: Array[Char], offset: Int, len: Int): Int = {
+    var i = offset
+    var hash = 0
+    while (i < len + offset) {
```

Comment:
Compute `len+offset` only once.

Reply:
Pretty sure the jvm should figure it out :).